### PR TITLE
Add border configuration support in the new email editor [MAILPOET-5821]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

Because the new email editor started displaying border configuration for blocks after merging the `core/button` renderer, we needed to implement the correct rendering for the other block as well.

## Code review notes

- For the proper rendering in various email clients and compatibility with mobile clients, I needed to use the `box-sizing: border-box` style that is not supported in each client. It seemed like the best compromise to me. As a next step, I added a warning to the border configuration panel. For this purpose, I used the `Fill` component from the WP components package.
- The new component for rendering warnings is placed behind `InterfaceSkeleton`, which allows the message to be displayed at the end of the specific slot. We could put it before and show the warning in the beginning. This is something that can be discussed.
- There is a known code reputation that I would like to address in the rendering refactor ticket.
- The whole implementation can look more complex due to support across many email clients.

## QA notes

- I tried to test the output via Email on Acid, and I will keep the output here for some time.
- I think we can skip QA here or let the reviewew do at least basic testing.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5821]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5821]: https://mailpoet.atlassian.net/browse/MAILPOET-5821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Column.php [hunk 1]
```diff
@@ -23,6 +23,24 @@ public function render(string $blockContent, array $parsedBlock, SettingsControl
...
+    $borderRightColor = $parsedBlock['attrs']['style']['border']['right']['color'] ?? $borderColor;
+    $borderTopColor = $parsedBlock['attrs']['style']['border']['top']['color'] ?? $borderColor;
+    $borderBottomWidth = $parsedBlock['attrs']['style']['border']['bottom']['width'] ?? $borderWidth;
+    $borderLeftWidth = $parsedBlock['attrs']['style']['border']['left']['width'] ?? $borderWidth;
+    $borderRightWidth = $parsedBlock['attrs']['style']['border']['right']['width'] ?? $borderWidth;
+    $borderTopWidth = $parsedBlock['attrs']['style']['border']['top']['width'] ?? $borderWidth;
+    $borderBottomLeftRadius = $parsedBlock['attrs']['style']['border']['radius']['bottomLeft'] ?? $borderRadius;
+    $borderBottomRightRadius = $parsedBlock['attrs']['style']['border']['radius']['bottomRight'] ?? $borderRadius;
+    $borderTopLeftRadius = $parsedBlock['attrs']['style']['border']['radius']['topLeft'] ?? $borderRadius;
+    $borderTopRightRadius = $parsedBlock['attrs']['style']['border']['radius']['topRight'] ?? $borderRadius;
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Column.php]: We discussed it on a call and agreed to utilize `wp_style_engine_get_styles` and replace this logic in all block renders with something like:

`$borderStyles = wp_style_engine_get_styles(['border' => $parsedBlock['attrs']['style']['border'] ?? []]);`

**Review by costasovo — COMMENTED** [review]
Let's try using wp_style_engine_get_styles

**Review by costasovo — APPROVED** [review]
@lysyjan Thank you for the update. I'm happy we got rid of some logic by using the WP's style engine function. I'm approving it. Please decide if you want to pass it to QA.

**Comment by lysyjan** [comment]
@costasovo I will merge it. If you agree, I will ask QA for a rendering test after solving all related tickets.

**Pull request merged**